### ISSUE TITLE
Introduce a UI option to "log design-time builds"

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn-project-system#1602]
In the new project system, similar to normal build warnings and errors, [design-time build](https://github.com/dotnet/roslyn-project-system/blob/master/docs/design-time-builds.md) warnings and errors show up in the Error List. The idea here is for a few different reasons:

1. It lets users a little more aware of the sorts of issues that break the design-time build, and hence the language service and other features that rely on it.

2. It lets the project system get of the business of logging specific errors and warnings when references can't be found, are the wrong framework, etc. Instead, we just surface the actual MSBuild warnings/errors that come from the build.

This is great, however, when you get an error/warning there's not an easy way to turn on logging for these builds to figure out _why_ these errors or warnings were logged. CPS does log these builds to the Output window if a specific registry key has been set - however, given the changes in the new installer to support side-by-side VS versions, this registry key now has be set in the privateregistry.bin in %APPDATA%. This is different to set, and has be set every new VS instance that you install.

To simplify this, I propose we add a new option in **Projects and Solutions** -> **Build and Run** to set this registry key:

![image](https://cloud.githubusercontent.com/assets/1103906/23245965/7dc3b1e6-f9e4-11e6-8f8a-de7ec4747e9c.png)

Optionally, we could also generalize this to turn on [TRACEDESIGNTIME](https://github.com/dotnet/roslyn-project-system/blob/master/docs/design-time-builds.md#visual-studio-2015-or-below) option for the legacy project system.

**Comment by KirillOsenkov** [comment]
Consider also adding this option:
Checkbox: `Log last 30 design-time builds to files`
Label: `Location` Textbox [...] (browse for folder button that lets you select a folder for logs, default to %temp%\VisualStudio\DesignTimeBuildLogs"

For each project being built in design-time, save the .binlog file to that folder. Only keep 30 most recent log files in that folder.

Since MSBuild is now getting BinaryLogger in the box:
https://github.com/Microsoft/MSBuild/wiki/Binary-Log

It should be easy to instantiate and pass the logger.

**Comment by bording** [comment]
I've been having to debug design-time builds a lot recently, so being able to to enable/disable the design-time build output window from a Visual Studio option would be a much nicer experience. 

I'd also like to be able to adjust the output verbosity separately from the regular build output.

**Comment by davkean** [comment]
C++/CPS also wanted a separate verbosity in an offline thread I had with them.

**Comment by tmat** [comment]
In general i'd like to have binlog option for build from VS (verbosity = "Minimal with binary log").

![image](https://user-images.githubusercontent.com/41759/53056118-f3996f80-345e-11e9-96c2-ab20fe75a12e.png)

**Comment by davkean** [comment]
That would fit under build log file verbosity, right? (Which C# doesn't respect right now)

**Comment by tmat** [comment]
Yes, that'd be reasonable. As long as I can get a binary log I don't care much what the exact setting is. It would be useful if the build output window printed out the path to the log file once the build is finished, so I don't need to hunt for it.

**Comment by dcorbin-wintellect** [comment]
4+ years later and crickets....

**Comment by drewnoakes** [comment]
@dcorbin-wintellect you might like to check out https://github.com/dotnet/project-system-tools
